collect_app/src/main/java/org/odk/collect/android/widgets/BarcodeWidget.java

```diff
@@ -94,12 +96,22 @@ public IAnswerData getAnswer() {
     @Override
     public void setData(Object answer) {
         String response = (String) answer;
+
         binding.barcodeAnswerText.setText(stripInvalidCharacters(response));
-        binding.barcodeAnswerText.setVisibility(binding.barcodeAnswerText.getText().toString().isBlank() ? GONE : VISIBLE);
         binding.barcodeButton.setText(getContext().getString(org.odk.collect.strings.R.string.replace_barcode));
+        updateVisibility();
+
         widgetValueChanged();
     }
 
+    private void updateVisibility() {
+        if (hasAppearance(getFormEntryPrompt(), Appearances.NO_ANSWER)) {
```

Comment:
If we do not display the answer in the widget we should not display it in the hierarchy view as well I think.

Reply:
Good point. I feel like it's ok to leave the data in the hierarchy right now as we can improve this later if people need it to hide there as well.